deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sDeployerUtil.java

```diff
@@ -41,6 +43,37 @@
 
     private static final Logger logger = LoggerFactory.getLogger(K8sDeployerUtil.class);
 
+    public static List<V1Pod> filterPods(ApiClient apiClient, String appName, String serviceName,
+            String namespace, List<V1Pod> podList) {
+        
+        if (podList == null || podList.isEmpty()) {
+            return podList;
+        }
+        
+        if (!PodPredicates.isPodAmbiguous().test(podList)) {
+            return podList;
+        }
```

Comment:
Combine these 2 conditions

Reply:
I kept them separate as this would be much more readable